### PR TITLE
feat(#5) - Add docker support

### PR DESCRIPTION
**Pull request by fabioalmeida100**
This PR is for add Docker support in back-end. 

**What was done in this PR?**

- Added file Dockerfile for make docker image
- Added file .dockerignore for ignore the folder **node_modules** at moment of build image
- Rewrote the section for setup of back-end in file README.md

### CODE REVIEW

**Comment by fabioalmeida100** [comment]
@Mayank0255 please, open the same issue in front-end (for setup Dockerfile) and assign to me. The Dockerfile is set up per project and as the project is separate now, is need a Dockerfile for back and other for front.

**Comment by Mayank0255** [comment]
> @Mayank0255 please, open the same issue in front-end (for setup Dockerfile) and assign to me. The Dockerfile is set up per project and as the project is separate now, is need a Dockerfile for back and other for front.

Sure, will do that and let you know so that you can comment on it which enables me to assign it to you,

And another thing that #10 has to merged first right? or the docker support has to be added first?

**Comment by Mayank0255** [comment]
@fabioalmeida100 I have created the issue for docker support in the frontend [this](https://github.com/Mayank0255/Stackoverflow-Clone-Frontend/issues/59)

**Comment by fabioalmeida100** [comment]
> @fabioalmeida100 I have created the issue for docker support in the frontend [this](https://github.com/Mayank0255/Stackoverflow-Clone-Frontend/issues/59)

Ok. Commented @Mayank0255

**Comment by fabioalmeida100** [comment]
> > @Mayank0255 please, open the same issue in front-end (for setup Dockerfile) and assign to me. The Dockerfile is set up per project and as the project is separate now, is need a Dockerfile for back and other for front.
> 
> Sure, will do that and let you know so that you can comment on it which enables me to assign it to you,
> 
> And another thing that #10 has to merged first right? or the docker support has to be added first?

No problem if you merge #10 first or later, because this PR no depends on the other 👍

**Comment by Mayank0255** [comment]
@fabioalmeida100 Understood, so what I will do is merge the docker changes to the master directly and for the sequelize thing, can you redirect it to a new branch `feature/sequelize`? I will just make that branch

**Review by Mayank0255 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -68,9 +71,27 @@ This repo consists of the Backend code of the project, the backend code is in [S
...
+    ```
+    docker run -p -d 3000:5000 stackoverflowclone
+    ```
 
-_Follow the steps properly and you are good to go._
+The default port of api is 5000. After running the container, you can access the api by typing:
+        
+    http://localhost:3000/api/<endpoint that you request - see next section>
 
+_Follow the steps properly (manual or Docker) and you are good to go._
```
Comment (Mayank0255) [inline README.md:94]: @fabioalmeida100 Do we not need to run the mysql scripts now?

**Comment by Mayank0255** [comment]
@fabioalmeida100 
My `.env` consists of this:
```
HOST = 'localhost'
USER = 'root'
DATABASE = 'stack_overflow_v2'
PASSWORD = 'Mayank15$'
```

This is what my mysql local db consists
image

Then I tried with a new db `stack_overflow_v3`, in which the local setup of the wasn't done, then also I got the same error I showed you before

**Comment by Mayank0255** [comment]
By the way it is running as it should be with the other setup guidelines with this `.env`

**Comment by fabioalmeida100** [comment]
> @fabioalmeida100 My `.env` consists of this:
> 
> ```
> HOST = 'localhost'
> USER = 'root'
> DATABASE = 'stack_overflow_v2'
> PASSWORD = 'Mayank15$'
> ```
> 
> This is what my mysql local db consists image
> 
> Then I tried with a new db `stack_overflow_v3`, in which the local setup of the wasn't done, then also I got the same error I showed you before

Answered in PR #16

**Pull request merged**